### DISCUSSION TITLE
JavaScript connect

### DISCUSSION THREAD

**Discussion by YALouis** [nextgenhealthcare/connect#5775] [General]
why mirth use rhino to change JavaScript to Java，rather than use java directly. In that case ,I may implement some functions easier，because there are more resources of Java

**Comment by mjtrac** [comment]
Just in case you aren't aware, you can add any Java jars you want to your system, and have access to the message in Java.  

https://docs.nextgen.com/bundle/Mirth_User_Guide_42/page/connect/connect/topics/c_Using_Java_Classes_connect_ug.html

**Reply by YALouis** [reply]
yes,I know how to add jar to mirth ,but in this case,may I could only use the functions , not the variable declaration, even if the variable is generated using a special initialization function

**Reply by tonygermano** [reply]
I don't understand exactly what you are trying to do, but almost anything you can do in java can be done in javascript in mirth with a slight syntax change.
